### PR TITLE
refactor(behavior_velocity_no_stopping_area_module): cppcheck warnings

### PR DESCRIPTION
**Pull request by veqcc**
## Description

I have made several chages based on CppCheck warnings.
Some commits are irrelevant to CppCheck.

1. remove redundant if branch
https://github.com/autowarefoundation/autoware.universe/commit/e0e49be0b07240e9c2097ed726397badeb983dfd
```
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:94:13: style: Condition '!collision_points.empty()' is always true [knownConditionTrueFalse]
        if (!collision_points.empty()) {
            ^
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:90:35: note: Assuming that condition 'collision_points.empty()' is not redundant
        if (collision_points.empty()) {
                                  ^
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:94:13: note: Condition '!collision_points.empty()' is always true
        if (!collision_points.empty()) {
            ^
```

2. remove unused variable
https://github.com/autowarefoundation/autoware.universe/commit/021129ad8079f4c90443d6b2b6ee6c5ba0a208e2

3. remove redundant initialization
https://github.com/autowarefoundation/autoware.universe/commit/0eb34eb5470efc652051f3851fdea0fdf773671a
```
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:333:20: style: Redundant initialization for 'ego_area_end_idx'. The initialized value is overwritten before it is read. [redundantInitialization]
  ego_area_end_idx = ego_area_start_idx;
                   ^
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:309:27: note: ego_area_end_idx is initialized
  size_t ego_area_end_idx = ego_area_start_idx;
                          ^
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:333:20: note: ego_area_end_idx is overwritten
  ego_area_end_idx = ego_area_start_idx;
                   ^
```

4. remove unused dist_from_area_sum variable
https://github.com/autowarefoundation/autoware.universe/commit/9e76436c08597851c8143ed45cf07e53865e19c4
```
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:338:26: style: Variable 'dist_from_area_sum' is assigned a value that is never used. [unreadVariable]
      dist_from_area_sum += tier4_autoware_utils::calcDistance2d(pp.at(i), pp.at(i - 1));
                         ^
```

### Question
I have one question: there seems to be several unused functions in debug.cpp.
Is it intentional? If not, it looks it's better to delete them in terms of maintenance and readbility.
```
planning/behavior_velocity_no_stopping_area_module/src/debug.cpp:134:0: style: The function 'createDebugMarkerArray' is never used. [unusedFunction]
visualization_msgs::msg::MarkerArray NoStoppingAreaModule::createDebugMarkerArray()
^
planning/behavior_velocity_no_stopping_area_module/src/debug.cpp:165:0: style: The function 'createVirtualWalls' is never used. [unusedFunction]
motion_utils::VirtualWalls NoStoppingAreaModule::createVirtualWalls()
^
planning/behavior_velocity_no_stopping_area_module/src/manager.cpp:54:0: style: The function 'launchNewModules' is never used. [unusedFunction]
void NoStoppingAreaModuleManager::launchNewModules(
^
planning/behavior_velocity_no_stopping_area_module/src/manager.cpp:77:0: style: The function 'getModuleExpiredFunction' is never used. [unusedFunction]
NoStoppingAreaModuleManager::getModuleExpiredFunction(
^
planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:108:0: style: The function 'modifyPathVelocity' is never used. [unusedFunction]
bool NoStoppingAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason * stop_reason)
^
```

## Tests performed

No

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by shmpwk** [comment]
@veqcc 
Thank you for your OSS contribution in holiday!
Regarding your question about unused functions in debug.cpp, I agree to delete those functions. But let me make sure to ask @kosuke55 -san who seems to be a heavy user of no stopping area as a goal planner developper.

**Comment by kosuke55** [comment]
@veqcc 
> I have one question: there seems to be several unused functions in debug.cpp.

thanks! Is it ok to check only debug.cpp? 
I think manager.cpp's functions (`launchNewModules` etc) are used:thinking:

**Comment by veqcc** [comment]
@kosuke55 

> thanks! Is it ok to check only debug.cpp?
I think manager.cpp's functions (launchNewModules etc) are used 🤔

You are right :+1: 


> And other modules in behavior veclocity planner has same structure and fuctions. Do you have any idea why only no_stopping_area module got the unusedFunction error?

Actualy, there are similar `unusedFunction` warnings in other modules.
Since this is a PR for `behavior_velocity_no_stopping_area_module`, I mentioned only related warnings.
If you prefer not to delete these functions, I think it also makes sense.

**Comment by kosuke55** [comment]
@veqcc 
It seem to get this warning about functions that is overridden. And I checked function in debug.cpp is also used. So we need to keep those functions.

**Review by kosuke55 — APPROVED** [review]
LGTM!

**Comment by veqcc** [comment]
I will fix the build warning later

**Comment by veqcc** [comment]
@kosuke55 
Hi, I have one question on this commit:
https://github.com/autowarefoundation/autoware.universe/pull/6986/commits/9e76436c08597851c8143ed45cf07e53865e19c4

Here, I have deleted `dist_from_area_sum` variable since it is only update inside the first if branch but soon breaks the loop.
Consequently, the parqameter `margin` is not necessary anymore, and in tern the following two function calls become totally the same https://github.com/autowarefoundation/autoware.universe/blob/b6bd75073a61757b6212c578409363680df2ddd8/planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp#L153C3-L160C43 .
Is it correct?
```
const Polygon2d stuck_vehicle_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, ego_space_in_front_of_stuck_vehicle,
    planner_param_.detection_area_length);
const Polygon2d stop_line_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, ego_space_in_front_of_stop_line,
    planner_param_.detection_area_length);
↓
const Polygon2d stuck_vehicle_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, // margin is deleted
    planner_param_.detection_area_length);
const Polygon2d stop_line_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, // margin is deleted
    planner_param_.detection_area_length);
```

File: planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp [hunk 1]
```diff
@@ -87,12 +91,8 @@ boost::optional<LineString2d> NoStoppingAreaModule::getStopLineGeometry2d(
         const LineString2d line{{p0.x, p0.y}, {p1.x, p1.y}};
         std::vector<Point2d> collision_points;
         bg::intersection(area_poly, line, collision_points);
-        if (collision_points.empty()) {
-          continue;
-        }
-        const double yaw = tier4_autoware_utils::calcAzimuthAngle(p0, p1);
```
Comment (soblin) [inline planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:93]: it is possible that `collision_points.empty()` , so this check is necessary

**Review by soblin — CHANGES_REQUESTED** [review]

File: planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:93]: same check seems to be done in later line

image

**Comment by kosuke55** [comment]
@veqcc 
> Here, I have deleted dist_from_area_sum variable since it is only update inside the first if branch but soon breaks the loop.
Consequently, the parqameter margin is not necessary anymore, and in tern the following two function calls become totally the same https://github.com/autowarefoundation/autoware.universe/blob/b6bd75073a61757b6212c578409363680df2ddd8/planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp#L153C3-L160C43 .
Is it correct?

The change was made in the PR https://github.com/autowarefoundation/autoware.universe/pull/6517  here and I checked with @kaigohirao san. dist_from_area_sum was already no longer needed, so you can delete them.

**Comment by veqcc** [comment]
@kosuke55 
Thank you for your detailed investigation!!
I understand we can delete `dist_from_area_sum` from `generateEgoNoStoppingAreaLanePolygon` function!!

Now, my question is this:
The `generateEgoNoStoppingAreaLanePolygon` function does not need `margin` parameter anymore, and thus the following two variables `stuck_vehicle_detect_area` and `stop_line_detect_area` will become exactly the same. It is true?
```
  const Polygon2d stuck_vehicle_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, ego_space_in_front_of_stuck_vehicle,
    planner_param_.detection_area_length);
  const Polygon2d stop_line_detect_area = generateEgoNoStoppingAreaLanePolygon(
    *path, current_pose->pose, ego_space_in_front_of_stop_line,
    planner_param_.detection_area_length);
``
https://github.com/autowarefoundation/autoware.universe/blob/b6bd75073a61757b6212c578409363680df2ddd8/planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp#L153C3-L160C43

**Comment by veqcc** [comment]
@kosuke55 
I have reverted the commit about `dist_from_area_sum` and created a new issue about it https://github.com/autowarefoundation/autoware.universe/issues/7081 :+1:

**Review by kosuke55 — APPROVED** [review]
@veqcc 
We will work on the issue in other PR. 
LGTM for your PR. Thank you!

**Review by soblin — APPROVED** [review]

**Pull request merged**